units/github_status_api.go

```diff
@@ -0,0 +1,278 @@
+package units
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/evergreen-ci/evergreen"
+	"github.com/evergreen-ci/evergreen/model/build"
+	"github.com/evergreen-ci/evergreen/model/patch"
+	"github.com/google/go-github/github"
+	"github.com/mongodb/amboy"
+	"github.com/mongodb/amboy/job"
+	"github.com/mongodb/amboy/registry"
+	"github.com/mongodb/grip"
+	"github.com/mongodb/grip/logging"
+	"github.com/mongodb/grip/message"
+	"github.com/pkg/errors"
+	"golang.org/x/oauth2"
+)
+
+const (
+	githubStatusUpdateJobName = "github-status-update"
+	githubStatusError         = "error"
+	githubStatusFailure       = "failure"
+	githubStatusPending       = "pending"
+	githubStatusSuccess       = "success"
+
+	githubUpdateTypeBuild = "build"
+	githubUpdateTypePatch = "patch"
+)
+
+func init() {
+	registry.AddJobType(githubStatusUpdateJobName, func() amboy.Job { return makeGithubStatusUpdateJob() })
+}
+
+type githubStatusUpdateJob struct {
+	job.Base `bson:"job_base" json:"job_base" yaml:"job_base"`
+	logger   grip.Journaler
+	env      evergreen.Environment
+
+	FetchID    string `bson:"fetch_id" json:"fetch_id" yaml:"fetch_id"`
+	UpdateType string `bson:"update_type" json:"update_type" yaml:"update_type"`
+
+	owner    string
+	repo     string
+	prNumber int
+	ref      string
+
+	urlPath     string
+	description string
+	context     string
+	ghStatus    string
```

Comment:
I think my preference in these sorts of situations is to: 

- make it exported, which means it gets saved in the job, which means that retries can avoid duplicate work, and we get a lot more traceability
- avoid saving this state on the job defintion and just pass it around (probably via some other type), or just having it in the scope of the Run() method. (this probably makes all the business logic more functional and tested.)

Reply:
done